mteb/evaluation/evaluators/RegressionEvaluator.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import numpy as np
+from scipy.stats import kendalltau
+from sklearn.linear_model import LinearRegression
+from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
+
+from mteb.encoder_interface import Encoder
+
+from .Evaluator import Evaluator
+
+logger = logging.getLogger(__name__)
+
+
+class LinearRegressionEvaluator(Evaluator):
+    def __init__(
+        self,
+        sentences_train,
+        y_train,
+        sentences_test,
+        y_test,
+        task_name: str,
+        hf_split: str,
+        hf_subset: str,
+        fit_intercept: bool = True,
+        limit: int | None = None,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        if limit is not None:
+            sentences_train = sentences_train[:limit]
+            y_train = y_train[:limit]
+            sentences_test = sentences_test[:limit]
+            y_test = y_test[:limit]
+        self.sentences_train = sentences_train
+        self.y_train = y_train
+        self.sentences_test = sentences_test
+        self.y_test = y_test
+
+        self.hf_split = hf_split
+        self.hf_subset = hf_subset
+        self.task_name = task_name
+        self.fit_intercept = fit_intercept
+
+    def __call__(
+        self,
+        model: Encoder,
+        *,
+        encode_kwargs: dict[str, Any] = {},
+        test_cache: np.ndarray | None = None,
+    ) -> tuple[dict[str, float], np.ndarray]:
+        scores = {}
+        X_train = model.encode(
+            self.sentences_train,
+            model=model,
+            task_name=self.task_name,
+            hf_split="train",
+            hf_subset=self.hf_subset,
+            **encode_kwargs,
+        )
+        if test_cache is None:
+            X_test = model.encode(
+                self.sentences_test,
+                model=model,
+                task_name=self.task_name,
+                hf_split=self.hf_split,
+                hf_subset=self.hf_subset,
+                **encode_kwargs,
+            )
+            test_cache = X_test
+
+        linear_regression = LinearRegression(fit_intercept=self.fit_intercept)
```

Comment:
Since people could use another regression model, then I would probably just allow people to pass in an sklearn regressor.

See an example from v2 [here](https://github.com/embeddings-benchmark/mteb/blob/ba7bb7df3d0e4fa1cbca60f1ee1eff1887186172/mteb/evaluation/evaluators/ClassificationEvaluator.py#L37C9-L37C19)